src/core/imap.ts

```diff
@@ -51,7 +59,7 @@ export class IMAP {
         });
     }
 
-    search(criteria) {
+    search(criteria): Promise<number[]> {
```

Comment:
Not a huge deal but you're code style is a bit inconsistent, here you annotate the return type, for some other functions in the same class you don't.

Reply:
Okay, I have now refacted the imap file and installed @types/imap to provide the imap function interfaces.

 I noticed that I did not annotate the functions in the standard way. Now I annotated functions that have a promise function inside the promise